File: college-growth-platform/backend/app/schemas/assessment.py
from datetime import datetime
from typing import Any
from uuid import UUID
from pydantic import BaseModel, Field

class OptionSchema(BaseModel):
    label: str
    value: int

class AssessmentQuestionSchema(BaseModel):
    id: UUID
    type: str
    dimension: str | None = None
    question_text: str
    options: list[OptionSchema]
    sort_order: int

    class Config:
        from_attributes = True

class AssessmentBase(BaseModel):
    id: UUID
    type: str
    status: str
    current_question: int = 0
    total_questions: int = 0
    started_at: datetime | None = None
    completed_at: datetime | None = None
    created_at: datetime

    class Config:
        from_attributes = True

class AssessmentDetail(AssessmentBase):
    responses: list["AssessmentResponseSchema"] = []

class AssessmentResponseSchema(BaseModel):
    id: UUID
    question_id: UUID
    answer_value: Any

    class Config:
        from_attributes = True

class AnswerSubmit(BaseModel):
    question_id: UUID
    answer_value: Any

class AssessmentResultSchema(BaseModel):
    id: UUID
    type: str
    result_data: Any
    summary: str | None = None
    created_at: datetime

    class Config:
        from_attributes = True

class AssessmentHistorySchema(BaseModel):
    id: UUID
    type: str
    result_snapshot: Any
    version: int
    created_at: datetime

    class Config:
        from_attributes = True

class StartAssessmentResponse(BaseModel):
    assessment: AssessmentBase
    questions: list[AssessmentQuestionSchema]

class AssessmentListResponse(BaseModel):
    holland: AssessmentBase | None = None
    ability: AssessmentBase | None = None
    values: AssessmentBase | None = None
    learning_habit: AssessmentBase | None = None
    readiness: AssessmentBase | None = None

class SubmitResponse(BaseModel):
    assessment_id: UUID
    result: AssessmentResultSchema
    message: str = "测评提交成功"

File: college-growth-platform/frontend/src/types/assessment.ts
export type AssessmentType = 'holland' | 'ability' | 'values' | 'learning_habit' | 'readiness';
export type AssessmentStatus = 'not_started' | 'in_progress' | 'completed';

export interface Option {
  label: string;
  value: number;
}

export interface AssessmentQuestion {
  id: string;
  type: AssessmentType;
  dimension?: string;
  question_text: string;
  options: Option[];
  sort_order: number;
}

export interface Assessment {
  id: string;
  type: AssessmentType;
  status: AssessmentStatus;
  current_question: number;
  total_questions: number;
  started_at?: string;
  completed_at?: string;
  created_at: string;
}

export interface AssessmentResult {
  id: string;
  type: string;
  result_data: Record<string, number | string>;
  summary?: string;
  created_at: string;
}

export interface AssessmentHistory {
  id: string;
  type: string;
  result_snapshot: Record<string, number | string>;
  version: number;
  created_at: string;
}

export interface StartAssessmentResponse {
  assessment: Assessment;
  questions: AssessmentQuestion[];
}

export interface SubmitAnswerResponse {
  question_id: string;
  current_question: number;
  total_questions: number;
  is_last: boolean;
}

export interface SubmitResponse {
  assessment_id: string;
  result: AssessmentResult;
  message: string;
}

export interface AssessmentStatusMap {
  holland: Assessment | null;
  ability: Assessment | null;
  values: Assessment | null;
  learning_habit: Assessment | null;
  readiness: Assessment | null;
}

export const ASSESSMENT_TYPES: AssessmentType[] = [
  'holland', 'ability', 'values', 'learning_habit', 'readiness'
];

export const ASSESSMENT_LABELS: Record<AssessmentType, string> = {
  holland: '霍兰德兴趣测评',
  ability: '能力自评',
  values: '职业价值观测评',
  learning_habit: '学习习惯测评',
  readiness: '准备度测评',
};

export const ASSESSMENT_DESCRIPTIONS: Record<AssessmentType, string> = {
  holland: '探索你的职业兴趣类型，找到最适合的发展方向',
  ability: '评估你在学习、沟通、协作等多维度的能力水平',
  values: '了解你内心最看重的职业价值观',
  learning_habit: '分析你的学习习惯，发现改进空间',
  readiness: '评估你对未来职业或深造的准备程度',
};

export const ASSESSMENT_ICONS: Record<AssessmentType, string> = {
  holland: '🧭',
  ability: '💪',
  values: '⭐',
  learning_habit: '📚',
  readiness: '🎯',
};
